From 957ebce3837588f71016e37ffaf9aad0a9d41cec Mon Sep 17 00:00:00 2001
From: Tanu Kaskinen <tanuk@iki.fi>
Date: Fri, 23 Sep 2016 12:02:06 +0300
Subject: [PATCH] configure.ac: improve alsa handling

Three improvements:

1) "--enable-alsa" or "--disable-alsa" caused a warning about an
unsupported configure option, because AC_ARG_ENABLE was not used.

2) If alsa was disabled, the "Have ALSA" item in the summary would print
an empty string instead of "no".

3) If "--enable-alsa" was passed to configure, but the headers were not
found, configure would still succeed (with alsa disabled). It's better
to fail and abort configure if a feature that was explicitly requested
can't be enabled.

Upstream-Status: Submitted [sent to src@mega-nerd.com]

Signed-off-by: Tanu Kaskinen <tanuk@iki.fi>
---
 configure.ac | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 474c6ab..bad597e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -101,6 +101,9 @@ AC_ARG_ENABLE(cpu-clip,
 		AC_HELP_STRING([--disable-cpu-clip], [disable tricky cpu specific clipper]),
 		ac_arg_cpu_clip="N", ac_arg_cpu_clip="Y")
 
+AC_ARG_ENABLE(alsa,
+                AC_HELP_STRING([--disable-alsa], [disable ALSA support in the varispeed-play example program]))
+
 #====================================================================================
 # Check types and their sizes.
 
@@ -194,6 +197,10 @@ if test x$enable_alsa != xno ; then
 	if test x$ac_cv_header_alsa_asoundlib_h = xyes ; then
 		ALSA_LIBS="-lasound"
 		enable_alsa=yes
+        elif test x$enable_alsa = xyes ; then
+                AC_MSG_ERROR(["ALSA headers not found"])
+        else
+                enable_alsa=no
 		fi
 	fi
 
@@ -340,7 +347,7 @@ AC_MSG_RESULT([
     Have FFTW : ................... ${ac_cv_fftw3}])
 
 AC_MSG_RESULT([    Have libsndfile : ............. ${ac_cv_sndfile}])
-AC_MSG_RESULT([    Have ALSA : ................... ${ac_cv_header_alsa_asoundlib_h}
+AC_MSG_RESULT([    Have ALSA : ................... ${enable_alsa}
 ])
 
 AC_MSG_RESULT([  Installation directories :
-- 
2.9.3

